rskj-core/src/test/java/co/rsk/core/bc/BlockChainImplTest.java

```diff
@@ -297,6 +299,30 @@ public void addInvalidBlockOneBadPaidFees() {
     }
 
     @Test
+    public void addInvalidBlockBadStateRoot() {
```

Comment:
For some reason we weren't testing this, but the code worked just fine.

It makes me nervous to think we allow instances of the class `Block` that pass have these two checks: I think we should have an `UnsafeBlock` and a `Block` that we can use separately to be 100% sure we don't mix them up.

For now, having these checks in the `tryToConnect` method will be enough.